widget/popup_test.go

```diff
@@ -201,3 +201,48 @@ func TestModalPopUp_TappedSecondary(t *testing.T) {
 	assert.Equal(t, 1, len(test.Canvas().Overlays().List()))
 	assert.Equal(t, pop, test.Canvas().Overlays().List()[0])
 }
+
+func TestModalPopUp_Resize(t *testing.T) {
+	label := NewLabel("Hi")
+	win := test.NewWindow(NewLabel("OK"))
+	win.Resize(fyne.NewSize(80, 80))
+	pop := NewModalPopUp(label, win.Canvas())
+	defer test.Canvas().Overlays().Remove(pop)
+
+	assert.Less(t, pop.Content.Size().Width, 70)
+	assert.Less(t, pop.Content.Size().Height, 50)
+
+	pop.Resize(fyne.NewSize(70, 50))
+	assert.Equal(t, 70-theme.Padding()*2, pop.Content.Size().Width)
+	assert.Equal(t, 50-theme.Padding()*2, pop.Content.Size().Height)
+	assert.Equal(t, 80, pop.Size().Width) // these are 80 as the popUp must fill our overlay
+	assert.Equal(t, 80, pop.Size().Height)
+}
+
+func TestModalPopUp_Resize_Constrained(t *testing.T) {
+	label := NewLabel("Hi")
+	win := test.NewWindow(NewLabel("OK"))
+	win.Resize(fyne.NewSize(80, 80))
+	pop := NewModalPopUp(label, win.Canvas())
+	defer test.Canvas().Overlays().Remove(pop)
+
+	pop.Resize(fyne.NewSize(90, 100))
+	assert.Equal(t, 80, pop.Content.Size().Width)
+	assert.Equal(t, 80, pop.Content.Size().Height)
+	assert.Equal(t, 80, pop.Size().Width)
+	assert.Equal(t, 80, pop.Size().Height)
+}
+
+func TestModalPopUp_Resize_ConstrainedMin(t *testing.T) {
+	label := NewLabel("Hi")
+	win := test.NewWindow(NewLabel("OK"))
+	win.Resize(fyne.NewSize(80, 80))
+	pop := NewModalPopUp(label, win.Canvas())
+	defer test.Canvas().Overlays().Remove(pop)
+
+	win.Resize(fyne.NewSize(30, 20))
+	assert.Greater(t, pop.Content.Size().Width, 30)
+	assert.Greater(t, pop.Content.Size().Height, 20)
+	assert.Greater(t, pop.Size().Width, 30)
+	assert.Greater(t, pop.Size().Height, 20)
```

Comment:
We probably should assert the canvas size here, shouldn't we?

Reply:
I don't see why - canvas size is controlled by window, a whole different area of tests (driver...)

Reply:
Okay, my request was probably not the correct one.
Let's try this:
Every test of `popUp.Resize()` should assert the final `popUp.Size()` and the final `popUp.Content.Size()` with literal values (+/- `theme.Padding()`).